tests/equality/equality.cpp

```diff
@@ -0,0 +1,995 @@
+    libcellml::VariablePtr v2 = libcellml::Variable::create();
+
+    v1->setName("variable");
+    v2->setName("variable");
+
+    v1->setInitialValue(3.0);
+    v2->setInitialValue("3");
+
+    v1->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+    v2->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+
+    v1->setUnits("ampere");
+    v2->setUnits("ampere");
+
+    EXPECT_TRUE(v1->equal(v2));
+}
+
+TEST(Equality, variableEqualAllTestedConditionsVariant2)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::VariablePtr v3 = libcellml::Variable::create("initialValue");
+    libcellml::VariablePtr v4 = libcellml::Variable::create("initialValue");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("m_per_s");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("m_per_s");
+
+    v1->setInitialValue(v3);
+    v2->setInitialValue(v4);
+
+    v1->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+    v2->setInterfaceType(libcellml::Variable::InterfaceType::PUBLIC);
+
+    v1->setUnits(u1);
+    v2->setUnits(u2);
+
+    EXPECT_TRUE(v1->equal(v2));
+}
+
+TEST(Equality, variableNotEqualByUnits)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create("m_per_s");
+
+    v1->setUnits(u1);
+
+    EXPECT_FALSE(v1->equal(v2));
+}
+
+TEST(Equality, variableNotEqualByUnitsOther)
+{
+    libcellml::VariablePtr v1 = libcellml::Variable::create("variable");
+    libcellml::VariablePtr v2 = libcellml::Variable::create("variable");
+
+    libcellml::UnitsPtr u2 = libcellml::Units::create("m_per_s");
+
+    v2->setUnits(u2);
+
+    EXPECT_FALSE(v1->equal(v2));
+}
+
+TEST(Equality, unitsEqualByNameOnly)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("units");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("units");
+
+    EXPECT_NE(u1, u2);
+    EXPECT_TRUE(u1->equal(u2));
+    EXPECT_TRUE(libcellml::Units::equivalent(u1, u2));
+}
+
+TEST(Equality, unitsNotEqualByNameOnly)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsB");
+
+    EXPECT_NE(u1, u2);
+    EXPECT_FALSE(u1->equal(u2));
+    EXPECT_FALSE(libcellml::Units::equivalent(u1, u2));
+}
+
+TEST(Equality, unitsEqualAllTestedConditionsVariant1)
+{
+    libcellml::UnitsPtr u1 = libcellml::Units::create("unitsA");
+    libcellml::UnitsPtr u2 = libcellml::Units::create("unitsA");
+
+    libcellml::ImportSourcePtr is = libcellml::ImportSource::create();
+
+    u1->setImportReference("some_location");
+    u2->setImportReference("some_location");
+
+    u1->setSourceUnits(is, "ampere");
+    u2->setSourceUnits(is, "ampere");
+
+    u1->addUnit("meter");
```

Comment:
See comment on python tests too - NB will not resolve to built-in units.

Reply:
Sorry I don't understand this comment.

Reply:
The spelling of `meter` is wrong.  It should be `metre` if you want it to represent built in units.  If not, I suggest you change the name to something more obviously custom?